trin-types/src/node_id.rs

```diff
@@ -1,26 +1,21 @@
-use crate::enr::Enr;
-use serde::{Deserialize, Serialize};
-use std::ops::Deref;
-use stremio_serde_hex::{SerHex, StrictPfx};
-
-use discv5::enr::NodeId as EnrNodeId;
-
 use crate::distance::{Metric, XorMetric};
+use crate::enr::Enr;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use trin_utils::bytes::{hex_decode, hex_encode};
 
 type RawNodeId = [u8; 32];
 
 /// Discv5 NodeId
-// TODO: Wrap this type over discv5::NodeId type
-#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
-pub struct NodeId(#[serde(with = "SerHex::<StrictPfx>")] pub RawNodeId);
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub struct NodeId(pub discv5::enr::NodeId);
 
 impl NodeId {
     pub fn random() -> Self {
-        Self(rand::random())
+        Self(discv5::enr::NodeId::random())
```

Comment:
I prefer to see all type imports at the top of the file. I'd go with something like `use discv5::enr::NodeId as Discv5NodeId;`.  However, an exception would be something like line 53, where I think it's fine to have an in-line import inside `map_err(..)`